consensus/cw_rating_calculator/cw_rating_calculator.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __COMMON_CONCENSUS_CW_RATING_CALCULATOR_CW_RATING_CALCULATOR_H__
+#define __COMMON_CONCENSUS_CW_RATING_CALCULATOR_CW_RATING_CALCULATOR_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "consensus/tangle/tangle.h"
+#include "common/errors.h"
+#include "common/trinary/trit_array.h"
+#include "uthash.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum CWCalculationImplementation {
+    NoImplementation,
+    TopologicalSorting,
+};
+
+typedef struct cw_entry_t {
+    trit_array_p hash;
+    int64_t cw;
+    UT_hash_handle hh;
+} cw_entry_t;
+
+typedef cw_entry_t *cw_map_t;
+
+typedef struct cw_rating_calculator_base cw_rating_calculator_base_t;
+typedef struct cw_rating_calculator_t cw_rating_calculator_t;
+
+typedef struct {
+    // find_transactions_request
+    retcode_t (*cw_rating_calculate)(
+            const cw_rating_calculator_t* const, trit_array_p entry_point, cw_map_t ratings);
+
+} cw_calculator_vtable;
+
+struct cw_rating_calculator_base {
+    cw_calculator_vtable vtable;
```

Comment:
How often will this be accessed? As there should only be one calculator in use per instance of e.g. cIRI, do we really need a vtable?